cluster_pack/spark/spark_config_builder.py

```diff
@@ -20,20 +28,48 @@ def add_packaged_environment(ssb: SparkSession.Builder, archive: str):
         os.environ['PYSPARK_PYTHON'] = "./condaenv/bin/python"
         os.environ['PYSPARK_DRIVER_PYTHON'] = 'python'
 
+    if not _get_value(ssb, "spark.master") == "yarn":
+        _logger.info("Not running on yarn. Adding archive to spark.files")
+        _add_or_merge(ssb, "spark.files", f"{archive}")
+
 
 def add_editable_requirements(ssb: SparkSession.Builder):
     for requirement_dir in packaging.get_editable_requirements().values():
         py_archive = packaging.zip_path(requirement_dir)
         _add_archive(ssb, py_archive)
 
 
+def add_s3_params(ssb: SparkSession.Builder,  fs_args: Dict[str, Any] = {}):
+    ssb.config("spark.hadoop.fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem")
+    ssb.config("spark.hadoop.fs.s3a.path.style.access", "true")
+    if "client_kwargs" in fs_args:
+        ssb.config("spark.hadoop.fs.s3a.endpoint", fs_args["client_kwargs"]["endpoint_url"])

```

Comment:
Can we let the user specify ACCESS_KEY/SECRET_KEY here ?

Reply:
Done